### ISSUE TITLE
[Bug]: Files: Context menu for files appearing in wrong location

### ISSUE THREAD

**Issue by pagdot** [nextcloud/server#47970]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

This bug is a two part bug.

Part one:
Right clicking on a file shows the context menu to far above:
![image](https://github.com/user-attachments/assets/5d155958-31ed-4647-b5dc-d62c1cfb24a6)

Part two:
After doing part one, selecting the file and pressing the "..." for the selected files at the top. The context menu containing delete shows up at the selected file, instead of the cursor position:
![image](https://github.com/user-attachments/assets/7124482e-fcb0-4e6a-88b6-f4b7de6e3270)

This only happens if part one is executed before

### Steps to reproduce

1. Right click on a file
2. Select the file
3. Press "..." for the selected file

### Expected behavior

Context menu appearing close to the cursor

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

This happens at the nextcloud conf instance https://conf.nextcloud.com/ running version 30.0.0.14

I'm running Firefox 128.0.3

**Comment by melroy89** [comment]
You selected "None" on most of the questions, that isn't nice.

**Comment by pagdot** [comment]
Yes, sorry about that. I don't own or maintain the nextcloud instance in question, so I can't comment on them. From what I've experienced the issues appeared to be entirely frontend based so I'd guess that the backend details don't influence it in any way

**Comment by pagdot** [comment]
Tested on another Firefox installation and on a Chromium installation.

About part one: I think the issue is the fixed offset from the cursor to the lower edge of the context menu. Seems like this is wrong and may be also dependent on screen scaling.

About part two: Managed to reproduce it consistently also on the second Firefox installation and on Chromium. So seems like it is independent on the browser

I've also recorded it to better show the issue
[Peek 2024-09-16 16-27.webm](https://github.com/user-attachments/assets/d4e1fc97-deba-40cc-b362-cbc6901b465f)

**Comment by arnowelzel** [comment]
Same here with NC 30.0.1 (in all instances - I run Nextlcoud on multiple servers). It also happens in all browsers (Chrome, Firefox, Edge, Vivaldi) I have teste. The context menu is too high when there is not enough space below the position of the mouse.

This is OK - the folder "Collectives" was clicked with the right mouse button:

![Image](https://github.com/user-attachments/assets/2be077b5-4367-48b4-88b9-52bb3b15f0f1)

And here you can see the problem - when you click a file near the lower border of the viewport, the context menu gets moved up too far - in this case, the file was "Reasons to use Nextcloud", but the menu appears too high:

![Image](https://github.com/user-attachments/assets/9e007242-2cca-4b0e-9a9f-472c95579758)

**Comment by Knusper** [comment]
Seeing this also since 30.0.1 - still appears in 30.0.2.  Firefox 132.0.1.
It only appears in the root folder.

**Comment by Felitendo** [comment]
Still happening on Nextcloud Hub 26 Spring (34.0.3):

The right click menu opens wayy to far above my cursor. I'm using a 4k Monitor with 150% Scaling on Windows with a Chromium Browser.

<img width="652" height="757" alt="Image" src="https://github.com/user-attachments/assets/87743053-027b-412f-ae3a-7055534ae9d0" />
